src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -440,8 +441,26 @@ private void delete(boolean cut, List<BibEntry> entries) {
             return;
         }
 
+        // Delete selected entries
         getUndoManager().addEdit(new UndoableRemoveEntries(bibDatabaseContext.getDatabase(), entries, cut));
         bibDatabaseContext.getDatabase().removeEntries(entries);
+
+        // Not delete files when user select cut
+        if (!cut) {
+            // Keep track of linked files from selected entries
+            List<LinkedFile> linkedFileList = new ArrayList<>();
+
+            for (BibEntry entry : entries) {
+                List<LinkedFile> linkedFilesFromEntry = entry.getFiles();
+                linkedFileList.addAll(linkedFilesFromEntry);
+            }
+
+            // Delete files which linked to selected entries when user select deletion
+            if (linkedFileList.size() > 0 && showLinkedFileDeleteConfirmationDialog(linkedFileList)) {
```

Comment:
Use `!....isEmpty()` instead of `.size() > 0` (this is what openrewrite also identfiees)